### DISCUSSION TITLE
Feature request: Turn Off Notifications for People I Follow

### DISCUSSION THREAD

**Discussion by jsjoeio** [forem/forem#15328] [Icebox]
**Is your feature request related to a problem? Please describe.**

I'm frustrated when I see notifications that are related to people I follow creating a new post.

**Describe the solution you'd like**

I'd love to be able to turn off notifications so that I am not notified when people I follow create a new post.

I'm not sure the best way to solve this but could be nice to have two options:

1. Turn off all notifications for posts by people I follow
2. Have a list of people I follow, and be able to selectively choose which people I want to be notified of their new posts.

This way, I only receive notifications when people interact with my posts. 
**Describe alternatives you've considered**

Current workarounds:
- Don't follow anyone
- Don't open notifications

(okay these are meant to be funny, but I don't see any real alternatives yet).

**Additional context**

I don't have time to work on it myself (or experience, no Ruby exp for me yet). 

Other people [on Twitter](https://twitter.com/jsjoeio/status/1139568483111276544?s=20) seem to agree this would be nice.

**Comment by maestromac** [comment]
Thank you for this issue @jsjoeio . The logic and the additional fields for this feature could sit in a couple of area; User Model, Follow Model, or the NotificationSubcription model. 

A rough idea of mine is to let NotificationSubscription decides whether or not to create a notification based on user's preference.

**Comment by jsjoeio** [comment]
Thank you for hearing my idea! @maestromac 🙏🏼

**Comment by thomasbnt** [comment]
Hello ! I find that this idea would harm the follow-up system. Let me explain: 

When you're a person, it's because you like what you're doing and you want to be there and notified when you post a new article. Unless we disable it for some people, I find that it breaks this system... it comes back a bit like those of big platforms (example: Facebook, Instagram, YouTube). 

Then I will see a general button that can disable notifications for X time or indefinitely.

**Comment by jsjoeio** [comment]
@thomasbnt hmm... I'm not sure I follow what you're saying. 

It sounds to me like you're mixing the "feed system" and the "notification system". 

On YouTube for example, I can subscribe to many channels to show I support them, but YouTube won't send me a notification every time those channels upload a new video - unless I specifically tell them to. 

Same thing with Facebook. I add a friend. Then I see their posts in my news feed, but I don't get a notification every time they post a new update (unless it's 2019 and FB is desperately trying to get my attention).

I don't believe my request is "harmful" - it gives more freedom/choice to the community members to chose what they're notified about. By default, it could stay the way it is. But for people like me who want to modify it, this feature would give us that option. Again, it's not meant to harm the system, but improve it. I hope that makes sense.

**Comment by thomasbnt** [comment]
Logically, when you are a person it is to be always up to date according to these posts, and therefore the current notifications are important and mark the value of the follow-up. 

The fact that YouTube, for example, have integrated the notification system with the bell, has clearly broken everything, because you have to follow the person but also activate if necessary, and in the other direction disable it if you don't want to. However, why not be notified if it is to follow the person and not specifically look at these posts?

**Comment by jsjoeio** [comment]
> Logically, when you are a person it is to be always up to date according to these posts, and therefore the current notifications are important and mark the value of the follow-up.

I understand what you're saying and agree there is indeed an important need there.

> However, why not be notified if it is to follow the person and not specifically look at these posts?

My impression was that following people === supporting them but I didn't expect to get notified every time they post. There are some people who I want to be notified when they post, but others who I don't. 

Maybe I am using dev.to incorrectly then.

What would be a good solution that fulfills both our needs without breaking the system?

**Comment by thomasbnt** [comment]
Maybe having a small list of people to follow is better than having too many notifications. As they always say, quality is better than quantity. Personally I prefer to receive notifications at each post to be always there to read them. 
 
> Maybe I am using dev.to incorrectly then.

I didn't say anything like that, so I think it's better to disable all notifications for a set time, as if to block apps and their notifications so as not to be distracted. It would also make it possible to read other authors.

**Comment by jsjoeio** [comment]
> Maybe having a small list of people to follow is better than having too many notifications. As they always say, quality is better than quantity.

This is true. I'll use this strategy for now.

👍🏼

**Comment by thomasbnt** [comment]
I'm not against being able to sort notifications for each person, I just deduce that it breaks the system and I wouldn't want dev.to to go through that... **so I optimize my subscriptions** and when I like a profile more I just unsubscribe :ok_hand:

**Comment by jessleenyc** [comment]
Just want to chime in here -- we _do_ want to give users the ability to fine-tune their notifications so if anyone wants to take on the original request of this issue, please go for it. 

@thomasbnt i do appreciate that you optimize your subscriptions though 😃

**Comment by twinsfan421** [comment]
I am interested in contributing and working on this.  I would like this feature as well.
I noticed #3240(Add backend functionality for muting users you follow) by adding a subscription_status column on the follow model.  I could use that for this feature request.

A few ideas I have for this:

I could add a (notifications on/noficications off) button next to the (follow/following) button that would update the subscription_status to "all_articles" or "none".  

If you wanted it under the users settings/notifications and to bulk set notifcations on/off, you could have maybe a Following User Notifications section with a receive notifications option that set off a background job to update all of the user's follows subscription_status. 
Or you could put the logic on the user model if you don't want that on the follow model. You could do a following_user_notifications similar to the email_badge_notifications. This would be ideal if you don't want the customization for each followed user but an option to turn notifications off/on for all of them.

@jessleenyc Would want this option on each followed user or a bulk turn on/off notifications for all or both?

**Comment by JennieOcken** [comment]
I'm gonna close this for now.  Lots of good discussion.  At some point we are going to come back and look at notifications with fresh eyes but not right now.

**Comment by michael-tharrington** [comment]
Just to say that I really like this idea!

It's come up in discussion recently [here](https://dev.to/tonymet/setting-for-bob-made-a-new-post-notification-3a7j) and I think it makes a lotta sense to give folks more control over their notifications.

Specifically, I'd like to see us add the following options under General Notifications here - https://dev.to/settings/notifications to give folks more control over these type of notifications:
> - Send notifications when someone I’m following posts an article
> - Send notifications when an organization I’m following posts an article

That said, I like @jsjoeio's idea for more granular controls:
> Have a list of people I follow, and be able to selectively choose which people I want to be notified of their new posts

Perhaps in a later iteration we could add that too!

In general, whenever we have space to update notifications, I think this would be a great addition! Perhaps we could even open this one up for community members to work on?

**Reply by tonymet** [reply]
thanks @michael-tharrington I'm https://dev.to/tonymet on dev.to .  

@jsjoeio summed up my expectations perfectly

> On YouTube for example, I can subscribe to many channels to show I support them, but YouTube won't send me a notification every time those channels upload a new video - unless I specifically tell them to.

**Comment by PhilipHow** [comment]
[Tracking here](https://github.com/forem/forem/issues/20533)

**Comment by tonymet** [comment]
I've created a draft PR for this.  the backend side is working well.  FE is in progress. I'll update the status there.

https://github.com/forem/forem/pull/20545

**Comment by tonymet** [comment]
@jsjoeio take a look at the PR it provides the functionality you're seeking

**Reply by jsjoeio** [reply]
nice! yeah that seems like a great fix! thanks for doing it!
